core-kotlin/src/main/java/com/amplifyframework/kotlin/datastore/DataStore.kt

```diff
@@ -47,6 +48,18 @@ interface DataStore {
     @Throws(DataStoreException::class)
     suspend fun <T : Model> delete(item: T, predicate: QueryPredicate = QueryPredicates.all())
 
+    /**
+     * Delete item(s) of a given class from the DataStore.
+     * @param byClass The class of item(s) being deleted
+     * @param filter Items must additionally match this filter, to be targeted for deletion.
+     *               If no filter is specified, an "allow all" predicate is used.
+     */
+    @Throws(DataStoreException::class)
+    suspend fun <T : Model> delete(
+        byClass: KClass<T>,
```

Comment:
Maybe name this variable `itemClass`, for consistency with the `query` method below?

Reply:
Done!